src/features/apiDelegation/offered/ChooseApiPage/ChooseApiPage.tsx

```diff
@@ -123,44 +150,62 @@ export const ChooseApiPage = () => {
         </div>
       );
     }
+
+    const prechosenApis = Array.from(urlParams.keys());
+
     return presentedApis.map((api: DelegableApi) => {
+      const initWithDelegationCheck = prechosenApis.includes(api.identifier);
       return (
-        <DelegationActionBar
+        <ApiActionBar
```

Comment:
Shoot, I didn't think of that...

Yeah,, the ActionBars won't be able to store that themselves unfortunately, since they're basically killed and reborn every time the search changes.

The only way of reminding them of their past lives is to store who've been previously selected in the parent, so a local state in ChooseApiPage that stores a list of the chosen but not delegable apis. We could technically do that and use the `initWithDelegationCheck` to trigger them refetcing the mutation from cache, but that would mean adding yet another state just for this and... idk.. I'm not sure if I think it's worth it honestly 😅

What do you think?